### PR TITLE
Display reason and status for Returned reviews

### LINKED ISSUE #576: Item returned for corrections displays wrong status in table
This item was submitted by the depositor for review, was reviewed and returned to the depositor for changes.

The Deposit Status in the table should say "Returned."

### LINKED ISSUE #581: Comments from Reviewer should be displayed at top of item edit page
In testing the review workflow, an item was returned for editing by the collection manager along with a comment.

When the depositor opened that item to make edits, the comment was not displayed on the page. https://sul-h2-qa.stanford.edu/works/38/edit.

### PR DESCRIPTION
**Pull request by ndushay**
Fixes #576
Fixes #581

## Why was this change made?

For #576:  display appropriate status for work rejected by reviewer

### After

image

For #581:  display comments from reviewer if work was rejected

### Before

image

### After

image

 
## How was this change tested?

Amy reviewed the changes for #581 in stage;   tests added for both issues with their commits.

### CODE REVIEW

File: app/components/dashboard/collection_component.html.erb
Comment (ndushay) [inline app/components/dashboard/collection_component.html.erb:32]: This is part of @mjgiarlo's efforts to normalize how we display state info.  He will have a PR forthcoming after _this_ PR is merged to finish this work.

File: app/components/works/state_display_component.rb
Comment (ndushay) [inline app/components/works/state_display_component.rb:11]: @amyehodge question as to whether you want this full string on the right to be displaying or not -- @mjgiarlo is hoping to normalize how we display state info.

File: app/models/work.rb
Comment (ndushay) [inline app/models/work.rb:73]: note that I specifically asked Amy to ensure that the rejection reason displayed at the appropriate times and that everything else seemed to work ok ...

File: app/views/dashboards/show.html.erb
Comment (ndushay) [inline app/views/dashboards/show.html.erb:32]: This is part of @mjgiarlo's efforts to normalize how we display state info.  He will have a PR forthcoming after _this_ PR is merged to finish this work.

File: config/locales/en.yml
Comment (jcoyne) [inline config/locales/en.yml:16]: Do we need both `deposited` and `depositing`?  Do we need `rejected` and `return`?

Comment (ndushay) [reply config/locales/en.yml:16]: I wish we didn't need them all but my testing found we did.  Perhaps the forthcoming "normalization of event/state display code text" PR will address?   Perhaps wiping the databases will address?  Dunno.  I didn't do the coding that required these -- I was only a consumer stymied by their absence.

Comment (jcoyne) [reply config/locales/en.yml:16]: I pulled this branch down and removed `return` and `depositing` and there were no test failures.  Perhaps this code needs coverage?

Comment (mjgiarlo) [reply config/locales/en.yml:16]: @jcoyne @ndushay I would like to issue a follow-up PR that tweaks the events attached to state transitions and how we label them. Maybe I could take this on?

File: spec/components/works/detail_component_spec.rb
Comment (mjgiarlo) [inline spec/components/works/detail_component_spec.rb:17]: good eye

**Review by mjgiarlo — APPROVED** [review]
LGTM, pending my comment about follow-up work.

File: config/locales/en.yml
Comment (ndushay) [reply config/locales/en.yml:16]: FYI, I think "depositing" is what you get locally when you hit deposit and the thing spins forever unless you manually give it a druid;  having an extra translation for that isn't a travesty, IMO.

File: app/components/works/state_display_component.rb
Comment (amyehodge) [reply app/components/works/state_display_component.rb:11]: I would say the "Not deposited" portion of the text does not need to display. And if I think about it, I wonder if that might make some people think that the "New version draft" somehow caused the original to be un-deposited? (not a thing, but they get very confused about the workflow sometimes, which is understandable).

Comment (mjgiarlo) [reply app/components/works/state_display_component.rb:11]: This will be changing per post-stand-up discussion. Stay tuned.

Comment (mjgiarlo) [reply app/components/works/state_display_component.rb:11]: @amyehodge for each of the six states here, could you write up the label you'd like to see in the app in a new issue, pls? It won't be a huge change, and it'd be good to get these right. Thanks!

**Pull request merged**
